### PR TITLE
Catch and report non HTTP/2 error case

### PR DESCRIPTION
**Pull request by mpl**

### CODE REVIEW

File: call.go [hunk 1]
```diff
@@ -236,6 +237,13 @@ func (cc *ClientConn) invoke(ctx context.Context, method string, args, reply int
 		if _, err := res.Body.Read(p.header[:1]); err != io.EOF {
 			return Errorf(codes.Internal, "grpc: malformed response with extra data after first message")
 		}
+	} else {
+		if len(p.header) == 0 {
+			// We received 0 bytes
+			if !res.ProtoAtLeast(2, 0) {
+				return Errorf(codes.Internal, fmt.Sprintf("grpc: HTTP/2 is required, but %v is used", res.Proto))
```
Comment (bradfitz) [inline call.go]: you don't need fmt.Sprintf. Errorf already does it.

Also, move this up to line 203-ish, right after you get the response

Comment (mpl) [reply call.go]: done

File: call.go [hunk 2]
```diff
@@ -199,6 +199,9 @@ func (cc *ClientConn) invoke(ctx context.Context, method string, args, reply int
 	}
 	defer res.Body.Close()
 
+	if !res.ProtoAtLeast(2, 0) {
+		return Errorf(codes.Internal, "grpc: HTTP/2 is required, but %v is used", res.Proto)
```
Comment (bradfitz) [inline call.go]: "grpc: HTTP/2 required; response was %v"

Comment (mpl) [reply call.go]: done

**Pull request merged**